src/main/java/ParkingLot/Vehicle.java

```diff
@@ -0,0 +1,107 @@
+package com.company;
+
+import java.util.Date;
+
+abstract public class Vehicle {
+    private String owner;
+    private String number;
+    private String type;
+    private Date dateIn;
+    private Date dateOut;
+    private int pricePerHour;
+    private int bill;
+    private int duration;
+
+    public Vehicle() {
+        this.dateIn = new Date();
+        this.dateOut = new Date();
+    }
+
+    public Vehicle(String owner, String number) {
+        this.owner = owner;
+        this.number = number;
+        this.dateIn = new Date();
+        this.dateOut = new Date();
+    }
+
+    abstract void Standard();
+
+    @Override
+    public String toString() {
+        return this.type + " number: " + this.number + ". Owner: " + this.owner + ". Check in: " + this.getDateIn(); //
+    }
+
+    public boolean checkIn() {
+        if (Main.freeLots >= 0) {
+            Date dateIn = new Date();
+            Main.freeLots = Main.freeLots - 1;
```

Comment:
change if-condition to Main.freeLots > 0 as now vehicle may check in even if there are 0 free lots

Reply:
done